### PR TITLE
feat: update all examples to allow token for SSO login users

### PR DESCRIPTION
**Pull request by ybyzek**
### Description 

1. Support for SSO login users do not have an email/password but they should be able to run the examples (i.e., anything programmatic). Note the new top-level `README.md` and `.env.defaults` file.
2. quickstart user creation now does not rely on developer login (it never should have)
3. quickstart `yarn start` presumes developer and workspace already created

@gwenshap  @scottydawg  even though the current developer auth token workflow is suboptimal because of the 24hr expiration, this PR takes the iterative approach: it enables SSO workflows and lays the groundwork for the workspace access token so we can substitute the developer auth token with minimal code changes. Therefore this PR is an improvement over what is there today (enables SSO users) and sets up for easy drop/replace when workspace access token is available.  

### Validated

- [x] quickstart
- [x] multi-tenancy
- [x] webapp

### CODE REVIEW

**Comment by ybyzek** [comment]
@scottydawg at this state, the PR updates the quickstart only.  I'd appreciate your review of the current set of changes before I modify all the other examples in this repo.

File: quickstart/src/entity-instance.ts [hunk 1]
```diff
@@ -6,32 +6,26 @@ import * as dotenv from 'dotenv';
...
+if (!process.env.NILE_URL || !process.env.NILE_WORKSPACE) {
+  console.error(emoji.get('x'), `Error: missing variable NILE_URL or NILE_WORKSPACE in .env .  See .env.defaults for more info and copy it to .env with your values`);
+  process.exit(1);
+}
 const NILE_URL = process.env.NILE_URL!;
 const NILE_WORKSPACE = process.env.NILE_WORKSPACE!;
-const NILE_DEVELOPER_EMAIL = process.env.NILE_DEVELOPER_EMAIL!;
-const NILE_DEVELOPER_PASSWORD = process.env.NILE_DEVELOPER_PASSWORD!;
-const NILE_ENTITY_NAME = "DB";
+var nile;
```
Comment (scottydawg) [inline quickstart/src/entity-instance.ts:16]: nit: for things that will be instantiated once, and should never change, it's usually safer to declare them like `const nile!;`. You should still be able to assign it like you do in `createInstance()`, but attempts to change it after that would produce an error.

**Review by scottydawg — APPROVED** [review]
One suggestion around `nile` declarations, but looks good overall.

File: quickstart/src/entity-instance.ts (hunk 1 above)
Comment (ybyzek) [reply quickstart/src/entity-instance.ts]: Thanks @scottydawg

**Pull request merged**